src/test/java/au/com/acegi/xmlformat/XmlCheckPluginTest.java

```diff
@@ -0,0 +1,157 @@
+/*-
+ * #%L
+ * XML Format Maven Plugin
+ * %%
+ * Copyright (C) 2011 - 2019 Acegi Technology Pty Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package au.com.acegi.xmlformat;
+
+import static au.com.acegi.xmlformat.TestUtil.fileToString;
+import static au.com.acegi.xmlformat.TestUtil.stringToFile;
+import java.io.File;
+import java.io.IOException;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugin.logging.Log;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests {@link XmlCheckPlugin}.
+ */
+public class XmlCheckPluginTest {
+
+  private static final String EMPTY_FILE_NAME = "empty.xml";
+  private static final String EMPTY_TXT = "";
+  private static final String ERR_FILE_NAME = "error.xml";
+  private static final String ERR_TXT = "<xml> <hello> hello not closed! </xml>";
+  private static final String NO_CHG_TXT = "<xml> <leave-me-alone/> </xml>";
+  private static final String TO_CHG_FILE_NAME = "my.xml";
+  private static final String TO_CHG_TXT = "<xml> <hello/> </xml>";
```

Comment:
A lot of duplication from `XmlFormatPluginTest`, but if I extract a parent class Checkstyle/PMD don't like protected fields, and I need to add getters for everything.